src/Control/Carrier/Class.hs

```diff
@@ -43,3 +57,80 @@ instance Monoid w => Carrier (Writer w) ((,) w) where
   eff (Tell w (w', k))    = (mappend w w', k)
   eff (Listen m k)        = uncurry k m
   eff (Censor f (w, a) k) = let (w', a') = k a in (mappend (f w) w', a')
+
+
+-- transformers
+
+instance (Carrier sig m, Effect sig) => Carrier (Error e :+: sig) (Except.ExceptT e m) where
+  eff (L (Throw e))     = Except.throwE e
+  eff (L (Catch m h k)) = Except.catchE m h >>= k
+  eff (R other)         = Except.ExceptT $ eff (handle (Right ()) (either (pure . Left) Except.runExceptT) other)
+
+instance Carrier sig m => Carrier (Reader r :+: sig) (Reader.ReaderT r m) where
+  eff (L (Ask       k)) = Reader.ask >>= k
+  eff (L (Local f m k)) = Reader.local f m >>= k
+  eff (R other)         = Reader.ReaderT $ \ r -> eff (hmap (flip Reader.runReaderT r) other)
+
+newtype RWSTF w s a = RWSTF { unRWSTF :: (a, s, w) }
+  deriving (Functor)
+
+toRWSTF :: Monoid w => w -> (a, s, w) -> RWSTF w s a
+toRWSTF w (a, s, w') = RWSTF (a, s, mappend w w')
```

Comment:
This was a lot easier than trying to futz around with swapping triples. Also, did you know that there’s no `Functor` instance for `(,,)`? What a pain.

Reply:
Jesus, yeah, I remember that. That shit sucks. Probably wanna `INLINE` this bad boy just to be sure, though.